src/shared-components/dropdown/desktopDropdown.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+import useResetFocus from 'src/utils/accessibility/useResetFocus';
+
+import OffClickWrapper from '../offClickWrapper';
+import ChevronIcon from '../../svgs/icons/chevron-icon.svg';
+import {
+  DropdownContainer,
+  dropdownInputStyle,
+  IconContainer,
+  DropdownOptionsContainer,
+  DropdownOption,
+} from './style';
+
+import { OptionType } from './index';
+
+type DesktopDropdownProps = {
+  value?: string;
+  options: OptionType[];
+  currentOption: OptionType;
+  textAlign: 'left' | 'center';
+  closeDropdown: () => void;
+  onOptionClick: (event: any) => void;
+  onSelectClick: () => void;
+  isOpen: boolean;
+  optionsContainerMaxHeight?: string;
+};
+
+const DesktopDropdown = ({
+  value,
+  options,
+  textAlign,
+  currentOption,
+  closeDropdown,
+  onOptionClick,
+  onSelectClick,
+  isOpen,
+  optionsContainerMaxHeight,
+}: DesktopDropdownProps) => {
+  const { initialFocus, resetFocus } = useResetFocus<HTMLDivElement>();
+
+  return (
+    <OffClickWrapper
+      onOffClick={closeDropdown}
+      css={css`
+        width: 100%;
+      `}
+    >
+      <DropdownContainer textAlign={textAlign}>
+        <div
+          id="select-input-box"
+          onClick={onSelectClick}
+          onKeyDown={(event: React.KeyboardEvent) => {
+            if (event.key === 'Enter') {
+              onSelectClick();
+            }
+          }}
+          tabIndex={0}
+          aria-label="Open dropdown option"
+          aria-haspopup="listbox"
+          role="button"
+          ref={initialFocus}
+        >
+          <div css={dropdownInputStyle({ textAlign })}>
+            {currentOption && currentOption.label}
+          </div>
+          <IconContainer>
+            <ChevronIcon width={10} height={10} rotate={isOpen ? 90 : 0} />
+          </IconContainer>
+        </div>
+
+        <DropdownOptionsContainer
+          isOpen={isOpen}
+          optionsContainerMaxHeight={optionsContainerMaxHeight}
+          role="listbox"
+          aria-activedescendant={value}
+          tabIndex={isOpen ? 0 : -1}
+          aria-labelledby="select-input-box"
+        >
+          {options.map(option => {
+            const {
+              value: optionValue, disabled, label, ...rest 
+            } = option;
+
+            return (
+              <DropdownOption
```

Comment:
Did you play around at all with trying to achieve the (kind of) opposite of `useResetFocus`, but rather than resetting focus back to another element, we actually set the focus *ahead* to the first descendent? (Kind of like the menu button when using keyboard navigation here: https://reach.tech/menu-button/#reach-skip-nav). 

I also have not dug into the internals of all this as much as you have, so I'm not use if that kind of functionality is consistent with "best practices" or we'd be potentially introducing hacky behavior that we ought to achieve with other code.

Reply:
This is a good idea - I'm going to write up a separate ticket for this and will investigate further. Looking at the `reach-ui` implementation, it looks like they generally send focus to the first list element, but it may make sense to focus on the selected element if there is one.